src/Catalyst.Core.Modules.P2P.Discovery.Hastings.Tests/UnitTests/DiscoveryTestBuilder.cs

```diff
@@ -0,0 +1,327 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Concurrency;
+using System.Reflection;
+using Catalyst.Abstractions.IO.Messaging.Correlation;
+using Catalyst.Abstractions.Network;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Abstractions.P2P.Discovery;
+using Catalyst.Abstractions.P2P.IO;
+using Catalyst.Abstractions.P2P.IO.Messaging.Correlation;
+using Catalyst.Abstractions.Util;
+using Catalyst.Core.Lib.IO.Messaging.Correlation;
+using Catalyst.Core.Lib.P2P.IO.Observers;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Lib.P2P.ReputationSystem;
+using Catalyst.Core.Lib.Util;
+using Catalyst.Protocol.Peer;
+using Catalyst.TestUtils;
+using Microsoft.Extensions.Caching.Memory;
+using NSubstitute;
+using Serilog;
+
+namespace Catalyst.Core.Modules.P2P.Discovery.Hastings.Tests.UnitTests
+{
+    public sealed class DiscoveryTestBuilder : IDisposable
+    {
+        private bool _autoStart;
+        private int _burnIn;
+        private ICancellationTokenProvider _cancellationProvider;
+        private IHastingsCareTaker _careTaker;
+        private IHastingsOriginator _currentState;
+        private IDns _dnsClient;
+        private int _hasValidCandidatesCheckMillisecondsFrequency;
+        private ILogger _logger;
+        private IPeerClient _peerClient;
+        private IPeerMessageCorrelationManager _peerCorrelationManager;
+        private IPeerRepository _peerRepository;
+        private IPeerSettings _peerSettings;
+        private IScheduler _scheduler;
+        private int _timeout;
+        public IList<IPeerClientObservable> PeerClientObservables;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)